server/lib/definitions.js

```diff
@@ -4655,4 +4542,319 @@ exports.bot = {
+      exports.basic.UPGRADES_TIER_3 = [exports.single];
+  exports.basic.UPGRADES_TIER_2 = [exports.smash];
+      exports.smash.UPGRADES_TIER_3 = [exports.megasmash, exports.spike, exports.autosmash];
+
+  exports.helibasic.UPGRADES_TIER_2 = [exports.helisquad, exports.helicruiser, exports.heliflank]
+      exports.helisquad.UPGRADES_TIER_3 = [exports.heliproducer, exports.helimaster]
+      exports.helicruiser.UPGRADES_TIER_3 = [exports.heliracer, exports.triheli]
+      exports.heliflank.UPGRADES_TIER_3 = [exports.helitrapper, exports.autoheli]
+
+  exports.twin.UPGRADES_TIER_2 = [exports.double, exports.bent, exports.gunner, exports.hexa];
+      exports.twin.UPGRADES_TIER_3 = [exports.triple];
+      exports.double.UPGRADES_TIER_3 = [exports.tripletwin, exports.split, exports.autodouble, exports.bentdouble];
+      exports.bent.UPGRADES_TIER_3 = [exports.penta, exports.spread, exports.benthybrid, exports.bentdouble, exports.triple];
+      exports.gunner.UPGRADES_TIER_3 = [exports.autogunner, exports.nailgun, exports.auto4,exports.machinegunner];
+
+  exports.sniper.UPGRADES_TIER_2 = [exports.assassin, exports.hunter, exports.mini, exports.builder];
+      exports.sniper.UPGRADES_TIER_3 = [exports.bushwhack];
+      exports.assassin.UPGRADES_TIER_3 = [exports.ranger, exports.falcon];
+      exports.hunter.UPGRADES_TIER_3 = [exports.preda, exports.poach, exports.sidewind];
+      exports.builder.UPGRADES_TIER_3 = [exports.construct, exports.autobuilder, exports.engineer, exports.boomer];
+
+  exports.machine.UPGRADES_TIER_2 = [exports.destroy, exports.artillery, exports.mini, exports.gunner];
+      exports.machine.UPGRADES_TIER_3 = [exports.spray];
+      exports.destroy.UPGRADES_TIER_3 = [exports.anni, exports.hybrid, exports.construct, exports.shotgun2];
+      exports.artillery.UPGRADES_TIER_3 = [exports.mortar, exports.spread, exports.skimmer];
+      exports.mini.UPGRADES_TIER_3 = [exports.stream, exports.nailgun];
+
+  exports.flank.UPGRADES_TIER_2 = [exports.hexa, exports.tri, exports.auto3, exports.flanktrap];
+      exports.flank.UPGRADES_TIER_3 = [];
+      exports.tri.UPGRADES_TIER_3 = [exports.fighter, exports.booster, exports.falcon, exports.bomber, exports.autotri];
+      exports.hexa.UPGRADES_TIER_3 = [exports.octo, exports.hexatrap];
+      exports.auto3.UPGRADES_TIER_3 = [exports.auto5, exports.heavy3, exports.auto4];
+      exports.flanktrap.UPGRADES_TIER_3 = [exports.bushwhack, exports.guntrap, exports.fortress, exports.bomber];
+
+  exports.director.UPGRADES_TIER_2 = [exports.overseer, exports.cruiser, exports.underseer];
+      exports.director.UPGRADES_TIER_3 = [exports.factory];
+      exports.overseer.UPGRADES_TIER_3 = [exports.overlord, exports.overtrap, exports.overgunner];  
+      exports.underseer.UPGRADES_TIER_3 = [exports.necromancer];
+      exports.cruiser.UPGRADES_TIER_3 = [exports.carrier, exports.battleship, exports.fortress];
+
```

Comment:
Removed the comment syntax `/*` right here and `*/` down there to enable more upgrades.